content/topics/java-specific/functional-interface/_index.md

```diff
@@ -0,0 +1,70 @@
+---
+content_type: topic
+ready: true
+tags:
+- functions
+title: Java Lambda expressions and Functional Interface
+---
+
+## Lambda expressions
+
+Lambda expressions, introduced in Java 8, are concise and anonymous blocks of code that can take parameters and return values, similar to methods.
+
+## Functional Interface
+
+Functional Interfaces are called Single Abstract Method (SAM) interface. They can have many default and static methods but only one abstract method - this is why they are called SAM. They make it possible to create abstractions that multiple classes can use without copying and pasting code. This is especially helpful when developers need to create a complex abstraction with various methods and behaviors.
+
+By combining functional interfaces with lambda expressions, we can achieve smaller and cleaner code, eliminating the need for excessive boilerplate code.
+
+#### Functional Interface have four types
```

Comment:
level 4 heading after level 2 heading. This should be a level 3 heading instead.

Also, this heading seems to be in cave-man speak and the capitalisation is a bit inconsistent.